implementation-guide/Implementation Guide.md

```diff
@@ -0,0 +1,632 @@
+
+Alisha creates this certificate using the session information from both her sessions with Carol's Node and Ben's node. Once Alisha has the `desc:handoff-give` she signs it in a `desc:sig-envolope` with her node's private key from her session with Carol's Node (Node A's private key in the Node A <-> Node C session). It might seem unusual to sign this handoff give that's being sent to Ben's node with the key Alisha uses with Carol's Node, but we must remember it's Carol's node who must verify this signature and by doing this Node C is able to verify that it was in fact Alisha's Node which created this `desc:handoff-give`.
+
+### creating the `desc:handoff-receive` certificate
+
+![](./handoffs-3party4.webm)
+
+The receiver doesn't just get the `desc:handoff-give` and pass it along, it generates its own certificate which includes the signed `desc:handoff-give`. The receiver does this to include some additional information only the receiver has that the exporter will need to securely perform the handoff. The certificate the receiver creates is called a `desc:handoff-receive` and looks like this:
+
+```
+<desc:handoff-receive receiving-session  ; The ID of the receiver <-> exporter session
+                      receiving-side     ; The public key of the receiver in the receiver <-> exporter session
+                      handoff-count      ; positive integer
+                      signed-give>       ; desc:sig-envolope containing a desc:handoff-give received from the gifter
+```
+
+We can see the receiver adds some information, but most of it is provided already by the gifter and is included by including that signed certificate along with the `desc:handoff-receive` in the `signed-give` field. One piece of information that's included here is the `handoff-count`, each session has an inrecementing integer starting at zero which reflects how many handoff-receive's they've sent in the session. They also include an integer which reflects how many handoff receive's they've received (with them being the exporter) within the session. This is used to prevent replay attacks, the exporter must only provide the reference once to prevent replay attacks. The last thing to note here is this is also wrapped in a `desc:sig-envolope` with a signature, but this signature is created with the receiver's private key in the gifter <-> receiver session. Again, this might seem a bit strange for two reasons:
```

Comment:
```suggestion
We can see the receiver adds some information, but most of it is provided already by the gifter and is included by including that signed certificate along with the `desc:handoff-receive` in the `signed-give` field. One piece of information that's included here is the `handoff-count`, each session has an inrecementing integer starting at zero which reflects how many handoff-receive's they've sent in the session. They also include an integer which reflects how many handoff receive's they've received (with them being the exporter) within the session. This is used to prevent replay attacks, the exporter must only provide the reference once to prevent replay attacks. The last thing to note here is this is also wrapped in a `desc:sig-envelope` with a signature, but this signature is created with the receiver's private key in the gifter <-> receiver session. Again, this might seem a bit strange for two reasons:
```